File: backend/src/main/java/com/spil/salesorder/service/interfaces/SalesOrderService.java
package com.spil.salesorder.service.interfaces;

import com.spil.salesorder.dto.SalesOrderDTO;

import java.util.List;

public interface SalesOrderService {
    List<SalesOrderDTO> getAllOrders();
    SalesOrderDTO getOrderById(Long id);
    SalesOrderDTO createOrder(SalesOrderDTO salesOrderDTO);
    SalesOrderDTO updateOrder(Long id, SalesOrderDTO salesOrderDTO);
    void deleteOrder(Long id);
}

File: frontend/src/services/salesOrderService.js
import apiClient from './api'

export const salesOrderService = {
  getAllOrders: async () => {
    const response = await apiClient.get('/sales-orders')
    return response.data
  },

  getOrderById: async (id) => {
    const response = await apiClient.get(`/sales-orders/${id}`)
    return response.data
  },

  createOrder: async (orderData) => {
    const response = await apiClient.post('/sales-orders', orderData)
    return response.data
  },

  updateOrder: async (id, orderData) => {
    const response = await apiClient.put(`/sales-orders/${id}`, orderData)
    return response.data
  },

  deleteOrder: async (id) => {
    await apiClient.delete(`/sales-orders/${id}`)
  },
}
